gobblin-cluster/src/main/java/org/apache/gobblin/cluster/SleepingTask.java

```diff
@@ -64,19 +74,27 @@ public void run() {
         throw new IOException("File creation error: " + taskStateFile.getName());
       }
       long endTime = System.currentTimeMillis() + sleepTime * 1000;
-      while (System.currentTimeMillis() <= endTime) {
-        Thread.sleep(1000L);
-        log.warn("Sleeping for {} seconds", sleepTime);
-      }
+
+      AssertWithBackoff.create().maxSleepMs(1000).timeoutMs(Integer.MAX_VALUE).backoffFactor(1).
+          assertTrue(new Predicate<Void>() {
+            @Override
+            public boolean apply(@Nullable Void input) {
+              return System.currentTimeMillis() > endTime && !infiniteSleep;
```

Comment:
Nit: change the order of evaluation to: !infiniteSleep && System.currentTimeMillis() > endTIme